lambdas/data-migration1/src/rules.ts

```diff
@@ -0,0 +1,128 @@
+import Knex from 'knex';
+
+import DynamoDbSearchQueue from '@cumulus/aws-client/DynamoDbSearchQueue';
+import {
+  PostgresCollectionRecord,
+  PostgresProviderRecord,
+  PostgresRuleRecord,
+  getRecordCumulusId,
+  tableNames,
+} from '@cumulus/db';
+import { envUtils } from '@cumulus/common';
+import Logger from '@cumulus/logger';
+
+import { RecordAlreadyMigrated } from './errors';
+import { MigrationSummary } from './types';
+
+const logger = new Logger({ sender: '@cumulus/data-migration/rules' });
+const Manager = require('@cumulus/api/models/base');
+const schemas = require('@cumulus/api/models/schemas');
+
+/**
+ * Migrate rules record from Dynamo to RDS.
+ *
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ *   Record from DynamoDB
+ * @param {Knex} knex - Knex client for writing to RDS database
+ * @returns {Promise<number>} - Cumulus ID for record
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migrateRuleRecord = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  knex: Knex
+): Promise<void> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(dynamoRecord, schemas.rule);
+
+  const existingRecord = await knex<PostgresRuleRecord>('rules')
+    .where({ name: dynamoRecord.name })
+    .first();
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Rule name ${dynamoRecord.name} was already migrated, skipping`);
+  }
+
+  const collectionCumulusId = dynamoRecord.collection
+    ? await getRecordCumulusId<PostgresCollectionRecord>(
+      { name: dynamoRecord.collection.name, version: dynamoRecord.collection.version },
+      tableNames.collections,
+      knex
+    )
+    : undefined;
+  const providerCumulusId = dynamoRecord.provider
+    ? await getRecordCumulusId<PostgresProviderRecord>(
+      { name: dynamoRecord.provider },
+      tableNames.providers,
+      knex
+    )
+    : undefined;
```

Comment:
I know that I'm the one who most recently changed these lines, but looking again, I think we need to surround these in a try/catch or Promise.allSettled, because `getRecordCumulusId` right now throws an error if the record is not found.

Or, `getRecordCumulusId` itself could just return `undefined` instead of throwing an error if the record is not found. That would simplify the code but I'm not sure if it's better or not. What do you think?